themes/powerline/powerline.base.bash

```diff
@@ -41,6 +41,15 @@ function __powerline_user_info_prompt {
   [[ -n "${user_info}" ]] && echo "${user_info}|${color}"
 }
 
+function __powerline_terraform_prompt() {
+  local terraform_workspace=""
+
+  if [ -d .terraform ]; then
+    terraform_workspace="$(command terraform workspace show 2>/dev/null)"
+    [[ -n "${terraform_workspace}" ]] && echo "${TERRAFORM_CHAR}${terraform_workspace}|${TERRAFORM_THEME_PROMPT_COLOR}"
+  fi
+}
+
```

Comment:
The core detection logic here could be placed in `~/themes/base.theme.bash` which would then be usable by all prompts in the system.  Perhaps something like (untested) :

_base.theme.bash_
```
function terraform_workspace_prompt {
    if _command_exists terraform ; then
        if [ -d .terraform ]; then
            echo -e "$(terraform workspace show 2>/dev/null)"
        fi
    fi
}
```

_powerline.base.bash_
```
function __powerline_terraform_prompt() {
    local terraform_workspace=""

    terraform_workspace="$(terraform_workspace_prompt)"
    [[ -n "${terraform_workspace}" ]] && echo "${TERRAFORM_CHAR}${terraform_workspace}|${TERRAFORM_THEME_PROMPT_COLOR}"
  fi
}
```

Reply:
done